src/dmd/mars.d

```diff
@@ -1345,6 +1362,70 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
         return true;
     }
 
+    /**
+     * Trigger an error messsage about an invalid switch.
+     * If an optional supplemental message has been provided,
+     * it will be printed too
+     *
+     * Params:
+     *  p = 0 terminated string
+     *  s = supplemental help message listing the available options
+     */
+    void errorInvalidSwitch(const(char)* p, string availableOptions = null)
+    {
+        error("Switch `%s` is invalid", p);
+        if (availableOptions !is null)
+            errorSupplemental(Loc.initial, "%.*s", availableOptions.length, availableOptions.ptr);
+    }
+
+    enum CheckOptions { error, help, success }
+
+    /*
+    Checks whether the CLI options contains a valid argument or a help argument.
+    If a help argument has been used, it will set the `usageFlag`.
+    Returns:
+        `1` if a valid argument has been passed and it's not a help page
+        `0` if a help page has
+    */
+    CheckOptions checkOptions(const(char)* p, ref bool usageFlag, string missingMsg)
+    {
+        // Checks whether a flag has no options (e.g. -foo or -foo=)
+        if (*p == 0 || strcmp(p, "=") == 0)
+        {
+            .error(Loc.initial, "%.*s", missingMsg.length, missingMsg.ptr);
+            errors = true;
```

Comment:
doesn't calling `.error()` also set `errors` ?

Reply:
`errors` is a local variable of `tryMain`